clients/src/test/java/org/apache/kafka/clients/consumer/internals/metrics/AsyncConsumerMetricsTest.java

```diff
@@ -0,0 +1,198 @@
+        if (consumerMetrics != null) {
+            consumerMetrics.close();
+        }
+        metrics.close();
+    }
+
+    @Test
+    public void shouldMetricNames() {
+        // create
+        consumerMetrics = new AsyncConsumerMetrics(metrics, CONSUMER_GROUP_PREFIX);
+        HashSet<MetricName> expectedMetrics = new HashSet<>(Arrays.asList(
+            metrics.metricName("last-poll-seconds-ago", CONSUMER_METRIC_GROUP),
+            metrics.metricName("time-between-poll-avg", CONSUMER_METRIC_GROUP),
+            metrics.metricName("time-between-poll-max", CONSUMER_METRIC_GROUP),
+            metrics.metricName("poll-idle-ratio-avg", CONSUMER_METRIC_GROUP),
+            metrics.metricName("commit-sync-time-ns-total", CONSUMER_METRIC_GROUP),
+            metrics.metricName("committed-time-ns-total", CONSUMER_METRIC_GROUP)
+        ));
+        expectedMetrics.forEach(metricName -> assertTrue(metrics.metrics().containsKey(metricName), "Missing metric: " + metricName));
+
+        HashSet<MetricName> expectedConsumerMetrics = new HashSet<>(Arrays.asList(
+            metrics.metricName("time-between-network-thread-poll-avg", CONSUMER_METRIC_GROUP),
+            metrics.metricName("time-between-network-thread-poll-max", CONSUMER_METRIC_GROUP),
+            metrics.metricName("application-event-queue-size", CONSUMER_METRIC_GROUP),
+            metrics.metricName("application-event-queue-time-avg", CONSUMER_METRIC_GROUP),
+            metrics.metricName("application-event-queue-time-max", CONSUMER_METRIC_GROUP),
+            metrics.metricName("application-event-queue-processing-time-avg", CONSUMER_METRIC_GROUP),
+            metrics.metricName("application-event-queue-processing-time-max", CONSUMER_METRIC_GROUP),
+            metrics.metricName("unsent-requests-queue-size", CONSUMER_METRIC_GROUP),
+            metrics.metricName("unsent-requests-queue-time-avg", CONSUMER_METRIC_GROUP),
+            metrics.metricName("unsent-requests-queue-time-max", CONSUMER_METRIC_GROUP),
+            metrics.metricName("background-event-queue-size", CONSUMER_METRIC_GROUP),
+            metrics.metricName("background-event-queue-time-avg", CONSUMER_METRIC_GROUP),
+            metrics.metricName("background-event-queue-time-max", CONSUMER_METRIC_GROUP),
+            metrics.metricName("background-event-queue-processing-time-avg", CONSUMER_METRIC_GROUP),
+            metrics.metricName("background-event-queue-processing-time-max", CONSUMER_METRIC_GROUP)
+        ));
+        expectedConsumerMetrics.forEach(metricName -> assertTrue(metrics.metrics().containsKey(metricName), "Missing metric: " + metricName));
+
+        // close
+        consumerMetrics.close();
+        expectedMetrics.forEach(metricName -> assertFalse(metrics.metrics().containsKey(metricName), "Missing metric: " + metricName));
+        expectedConsumerMetrics.forEach(metricName -> assertFalse(metrics.metrics().containsKey(metricName), "Missing metric: " + metricName));
```

Comment:
The error message on test failure is incorrect here, right? Shouldn't it be:

```suggestion
        expectedMetrics.forEach(metricName -> assertFalse(metrics.metrics().containsKey(metricName), "Metric present after close: " + metricName));
        expectedConsumerMetrics.forEach(metricName -> assertFalse(metrics.metrics().containsKey(metricName), "Metric present after close: " + metricName));
```